atlasdb-config/src/test/java/com/palantir/atlasdb/factory/timelock/TimestampCorroboratingTimelockServiceTest.java

```diff
@@ -0,0 +1,159 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory.timelock;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.lock.v2.AutoDelegate_TimelockService;
+import com.palantir.lock.v2.IdentifiedTimeLockRequest;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.StartAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampRange;
+import com.palantir.timestamp.TimestampService;
+
+public class TimestampCorroboratingTimelockServiceTest {
+    private static final IdentifiedTimeLockRequest IDENTIFIED_TIME_LOCK_REQUEST = IdentifiedTimeLockRequest.create();
+
+    private TimelockService timelockService;
+    private GoBackInTimeTimestampService goBackInTimeTimestampService;
+
+    @Before
+    public void setUp() {
+        goBackInTimeTimestampService = new GoBackInTimeTimestampService();
+        timelockService = TimestampCorroboratingTimelockService.create(
+                getMockTimelockService(goBackInTimeTimestampService));
+    }
+
+    @Test
+    public void getFreshTimestampShouldFail() {
+        getIndividualTimestamps(20);
+        goBackInTimeFor(10);
+
+        assertThatThrownBy(timelockService::getFreshTimestamp)
+                .isInstanceOf(AssertionError.class)
+                .hasMessage(generateClocksWentBackwardsMessage(20, 11));
+    }
+
+    @Test
+    public void getFreshTimestampsShouldFail() {
+        getBatchedTimestamps(20);
+        goBackInTimeFor(10);
+
+        assertThatThrownBy(() -> timelockService.getFreshTimestamps(10))
+                .isInstanceOf(AssertionError.class)
+                .hasMessage(generateClocksWentBackwardsMessage(20, 11));
+    }
+
+    @Test
+    public void startAtlasDbTransactionShouldFail() {
+        timelockService.startAtlasDbTransaction(IDENTIFIED_TIME_LOCK_REQUEST);
+        goBackInTimeFor(1);
+
+        assertThatThrownBy(() -> timelockService.startAtlasDbTransaction(IDENTIFIED_TIME_LOCK_REQUEST))
+                .isInstanceOf(AssertionError.class)
+                .hasMessage(generateClocksWentBackwardsMessage(1, 1));
+    }
+
+    private static String generateClocksWentBackwardsMessage(long lowerBound, long freshTimestamp) {
+        return String.format(TimestampCorroboratingTimelockService.CLOCKS_WENT_BACKWARDS_MESSAGE,
```

Comment:
this isn't what you want. In particular, the reason you assert on messages is to make sure the result was right, whereas this does not achieve that.